Improvements to UCN dynamic client docs (#1086)
hazelcast/hz-docs: docs/modules/clusters/pages/ucn-dynamic-client.adoc

= Configure Client

> :description: Clients use the dynamic configuration API to create new, and update existing, {ucn} on the cluster.

Author: ```suggestion
:description: Clients use the dynamic configuration API to create and update {ucn} on the cluster.
```

See, this is why I always look forward to your reviews @rebekah-lawrence! Looking back at it now, I don't think the commas are necessary either, so I've removed them with this suggestion.

:page-enterprise: true
:page-beta: false

{description}

> == Dynamically creating {ucn}

Reviewer: ```suggestion
== Dynamically Creating {ucn}
```

You must create a _Client.java_ file containing the required code and save it to your configuration folder.

> NOTE: {ucn} must be statically enabled on the member side before clients are allowed to dynamically create {ucn}.

Reviewer 2: ```suggestion
NOTE: {ucn} must be enabled using static configuration on the member side before clients are allowed to dynamically create {ucn}. For further information on enabling {ucn}, see xref:clusters:ucn-enable.adoc[].
```

> The file will be similar to the following:

Reviewer 2: ```suggestion
The created file will be similar to the following:
```

Author: I'm fine with this change, but just an FYI that it's used in several other UCN documentation pages as well (it was already on this page).

        UserCodeNamespaceConfig namespaceConfig = new UserCodeNamespaceConfig("namespace_name"); <1>
        namespaceConfig.addClass(IncrementingEntryProcessor.class); <2>

        // dynamically update the namespace config
        client.getConfig().getNamespacesConfig().addNamespaceConfig(namespaceConfig); <3>
        // execute the entry processor - this map should already be configured to use our namespace
        client.getMap("map_name").executeOnKey("key", entryProcessor); <4>
    }
}
----
<1> Creates a `UserCodeNamespaceConfig` with a name of `namespace_name`
<2> Adds the defined class from the classpath to the `UserCodeNamespaceConfig`
<3> Gets the Hazelcast configuration from the client instance and adds the `UserCodeNamespaceConfig` to the `Config`, overwriting the existing config

> <4> Executes `IncrementingEntryProcessor` from the classpath on the key in an IMap that has been configured to use the correct Namespace

Reviewer 2: ```suggestion
<4> Executes `IncrementingEntryProcessor` from the classpath on the key in an IMap that has been configured to use the same Namespace
```
